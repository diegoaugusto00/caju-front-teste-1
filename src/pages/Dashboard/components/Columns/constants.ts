const REVIEW = 'REVIEW';
const APPROVED = 'APPROVED';
const REPROVED = 'REPROVED';

export {
    REVIEW,
    APPROVED,
    REPROVED,
}