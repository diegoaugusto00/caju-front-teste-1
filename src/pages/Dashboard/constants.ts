const STATUS_REVIEW = 'REVIEW';
const STATUS_APPROVED = 'APPROVED';
const STATUS_REPROVED = 'REPROVED';

const STATUS = {
  REVIEW: STATUS_REVIEW,
  APPROVED: STATUS_APPROVED,
  REPROVED: STATUS_REPROVED,
};

export {
  STATUS_REVIEW,
  STATUS_APPROVED,
  STATUS_REPROVED,
  STATUS,
};