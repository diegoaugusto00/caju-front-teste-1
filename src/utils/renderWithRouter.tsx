import { ReactNode } from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

interface WrapperProps {
  children: ReactNode;
}

/**
 * Renderiza um componente React com o contexto do React Router para testes.
 *
 * @param {React.ReactElement} children - O componente filho React a ser renderizado.
 * @param {string} [options.route='/'] - A rota inicial.
 * @returns {Object} O resultado da renderização com o Wrapper e o history.
 */
const renderWithRouter = (
  children: React.ReactElement,
  { route = "/" } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const Wrapper: React.FC<WrapperProps> = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(children, { wrapper: Wrapper }),
    history,
  };
};

export default renderWithRouter;
