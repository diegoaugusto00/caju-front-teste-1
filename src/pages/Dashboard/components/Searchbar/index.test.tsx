import { screen, fireEvent, waitFor } from "@testing-library/react";
import { SearchBar } from ".";
import routes from "~/router/routes";
import renderWithRouter from "~/utils/__tests__/renderWithRouter";
import { STATUS } from "../../constants";

describe("SearchBar Component tests", () => {
  it("should render without crashing", () => {
    const { container } = renderWithRouter(<SearchBar />);
    expect(container).toBeInTheDocument();
  });

  it("should call onSearch with correct values when typing in the CPF fiels", async () => {
    const mockOnSearch = jest.fn();
    renderWithRouter(<SearchBar onSearch={mockOnSearch} />);

    const cpfInput = screen.getByPlaceholderText("Digite um CPF válido");
    fireEvent.change(cpfInput, { target: { value: "123.456.789-01" } });

    await waitFor(() => {
      expect(mockOnSearch).toHaveBeenCalledWith("12345678901", "");
    });
  });

  it("should call onSearch with values when selecting a status", async () => {
    const mockOnSearch = jest.fn();
    renderWithRouter(<SearchBar onSearch={mockOnSearch} />);

    const statusSelect = screen.getByRole("combobox");
    fireEvent.change(statusSelect, { target: { value: STATUS.APPROVED } });

    await waitFor(() => {
      expect(mockOnSearch).toHaveBeenCalledWith("", STATUS.APPROVED);
    });
  });

  it("should clear the search fields when clicking the refresh button", () => {
    renderWithRouter(<SearchBar />);

    const cpfInput = screen.getByPlaceholderText("Digite um CPF válido");
    const statusSelect = screen.getByRole("combobox");
    const refreshButton = screen.getByLabelText("refetch");

    fireEvent.change(cpfInput, { target: { value: "123.456.789-01" } });
    fireEvent.change(statusSelect, { target: { value: STATUS.APPROVED } });
    fireEvent.click(refreshButton);

    expect(cpfInput).toHaveValue("");
    expect(statusSelect).toHaveValue("");
  });

  it("should call onSearch with cpf and status", async () => {
    const mockOnSearch = jest.fn();

    renderWithRouter(<SearchBar onSearch={mockOnSearch} />);

    const cpfInput = screen.getByPlaceholderText("Digite um CPF válido");
    const statusSelect = screen.getByRole("combobox");

    fireEvent.change(cpfInput, { target: { value: "123.456.789-01" } });
    fireEvent.change(statusSelect, { target: { value: STATUS.APPROVED } });

    await waitFor(() => {
      expect(mockOnSearch).toHaveBeenCalledWith("12345678901", STATUS.APPROVED);
    });
  });

  it("should navigate to new admission page when click in the new admission button", () => {
    const { history } = renderWithRouter(<SearchBar />);

    const newAdmissionButton = screen.getByText("Nova Admissão");
    fireEvent.click(newAdmissionButton);

    expect(history.location.pathname).toBe(routes.newUser);
  });
});
