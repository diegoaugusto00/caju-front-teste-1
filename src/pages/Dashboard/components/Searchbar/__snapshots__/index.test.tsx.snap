// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component tests should navigate to new admission page when clicking the new admission button 1`] = `
<div>
  <div
    class="sc-iGgVNO ddLifx"
  >
    <div
      class="sc-kAycRU iGyHMU"
    >
      <div>
        <label />
        <input
          class="sc-dcJtft etWJkp"
          maxlength="14"
          placeholder="Digite um CPF válido"
          value=""
        />
        <span
          style="font-size: 12px; color: red;"
        />
      </div>
      <div
        class="sc-imWZod beiGUh"
      >
        <select
          class="sc-jXbVAB gOLGmz"
        >
          <option
            disabled=""
            value=""
          >
            Selecione um status
          </option>
          <option
            value=""
          >
            Todos
          </option>
          <option
            value="REVIEW"
          >
            Pendente
          </option>
          <option
            value="APPROVED"
          >
            Aprovado
          </option>
          <option
            value="REPROVED"
          >
            Reprovado
          </option>
        </select>
      </div>
    </div>
    <div
      class="sc-gsFSjX ga-dmrm"
    >
      <button
        aria-label="refetch"
        class="sc-eqUzNf iebrMx"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M4 2a1 1 0 011 1v2.101a7.002 7.002 0 0111.601 2.566 1 1 0 11-1.885.666A5.002 5.002 0 005.999 7H9a1 1 0 010 2H4a1 1 0 01-1-1V3a1 1 0 011-1zm.008 9.057a1 1 0 011.276.61A5.002 5.002 0 0014.001 13H11a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0v-2.101a7.002 7.002 0 01-11.601-2.566 1 1 0 01.61-1.276z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <button
        class="sc-aYaIB dyJnol"
      >
        Nova Admissão
      </button>
    </div>
  </div>
</div>
`;
